Framework/src/TaskRunnerFactory.cxx

```diff
@@ -152,6 +157,32 @@ TaskRunnerConfig TaskRunnerFactory::extractConfig(const CommonSpec& globalConfig
   };
 }
 
+InputSpec TaskRunnerFactory::createTimerInputSpec(const CommonSpec& globalConfig, std::vector<std::pair<size_t, size_t>>& cycleDurations,
+                                                  const std::string& detectorName, const std::string& taskName)
+{
+  // This is to check that the durations are not below 10 seconds except when using a dummy database
+  auto dummyDatabaseUsed = globalConfig.database.count("implementation") > 0 && globalConfig.database.at("implementation") == "Dummy";
+  for (auto& [cycleDuration, period] : cycleDurations) {
+    if (!dummyDatabaseUsed && cycleDuration < 10) {
+      ILOG(Error, Support) << "Cycle duration is too short (" << cycleDuration << "), replaced by a duration of 10 seconds." << ENDM;
+      cycleDuration = 10;
+    }
+  }
```

Comment:
Since you are doing something in that loop only if `dummyDatabaseUsed` is true, why don't you not execute the loop at all in such case? ...and get rid of `!dummyDatabaseUsed` in the inner if statement?

Reply:
of course